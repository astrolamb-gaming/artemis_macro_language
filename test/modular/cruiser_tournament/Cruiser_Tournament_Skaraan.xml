<mission_data version="1.0">
  <imports>
    <import name="Cruiser_Tournament_templates.xml" />
  </imports>
  <start>
    <create type="enemy" x="50500.0" y="0.0" z="41500.0" angle="260" name="K61" raceKeys="Skaraan enemy" hullKeys="small Defiler" fleetnumber="22"/>
    <set_special name="K61" ability="Tractor"/>
    <set_special name="K61" ship="-1" captain="5" ability="Warp"/>
    <set_special name="K61" clear="yes" ability="Cloak"/>
    <set_special name="K61" clear="yes" ability="AntiMine"/>
    <set_special name="K61" clear="yes" ability="Drones"/>
    <set_special name="K61" clear="yes" ability="ShldDrain"/>
    <set_special name="K61" clear="yes" ability="ShldVamp"/>
    <set_special name="K61" clear="yes" ability="LowVis"/>
    <set_special name="K61" clear="yes" ability="Stealth"/>
  </start>
  <expand _template="tonnage" name="K61" race="Skaraan" hull="Defiler" />
</mission_data> 