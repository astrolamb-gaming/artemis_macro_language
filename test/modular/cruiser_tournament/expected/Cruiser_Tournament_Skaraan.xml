<mission_data version="1.0">
  <start>
    <create type="enemy" x="50500.0" y="0.0" z="41500.0" angle="260" name="K61" raceKeys="Skaraan enemy" hullKeys="small Defiler" fleetnumber="22"/>
    <set_special name="K61" ability="Tractor"/>
    <set_special name="K61" ship="-1" captain="5" ability="Warp"/>
    <set_special name="K61" clear="yes" ability="Cloak"/>
    <set_special name="K61" clear="yes" ability="AntiMine"/>
    <set_special name="K61" clear="yes" ability="Drones"/>
    <set_special name="K61" clear="yes" ability="ShldDrain"/>
    <set_special name="K61" clear="yes" ability="ShldVamp"/>
    <set_special name="K61" clear="yes" ability="LowVis"/>
    <set_special name="K61" clear="yes" ability="Stealth"/>
  </start>
  <event name="K61 Surrendered">
    <if_variable name="K61_surrendered" comparator="EQUALS" value="0"/>
    <if_object_property property="hasSurrendered" name="K61" comparator="EQUALS" value="1"/>
    <set_variable name="K61_surrendered" value="1"/>
    <set_object_property property="topSpeed" value="0.2" name="K61"/>
    <set_variable name="Tonnage" value="Tonnage+40+20"/>
    <big_message title="Skaraan Defiler surrendered" subtitle1="40 kilotons" subtitle2="+20 kt surrender bonus"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Skaraan Defiler K61 Surrendered"/>
  </event>
  <event name="K61 Destroyed">
    <if_variable name="K61_surrendered" comparator="EQUALS" value="0"/>
    <if_variable name="K61_dead" comparator="EQUALS" value="0"/>
    <if_not_exists name="K61"/>
    <set_variable name="K61_dead" value="1"/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <set_variable name="Tonnage" value="Tonnage+40"/>
    <big_message title="Skaraan Defiler K61 destroyed" subtitle1="40 kilotons" subtitle2=""/>
    <set_variable name="report_score" value="1"/>
    <set_timer name="report_timer" seconds="8"/>
    <log text="Skaraan Defiler K61 destroyed"/>
  </event>
</mission_data>